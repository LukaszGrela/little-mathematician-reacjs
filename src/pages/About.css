.about {
    padding-top: 1em;
    text-align: left;
    width: 300px;
    margin: 0 auto;
}
.about .title {
    font-size: 2em;
    line-height: 3em;
}
.about .acknowledgement {
    color: #757575;
}