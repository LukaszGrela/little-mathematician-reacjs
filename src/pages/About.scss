/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

@import '../sass/_variables.scss';
.about {
    padding: 1em;
    text-align: left;
    width: $width-game-column;
    margin: 0 auto;
    .title {
        font-size: 2em;
        line-height: 3em;
    }
    .acknowledgement {
        color: $secondary-text-color;
    }
    .git-source {
        padding: 1rem 0;
        .git-hub-icon {
            vertical-align: bottom;
            width: 1.4rem;
            height: 1.4rem;
            padding-right: 0.4rem;
        }
    }
    
}