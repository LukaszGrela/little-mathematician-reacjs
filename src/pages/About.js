/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

import React, { Component } from 'react';

import './About.scss'
import GitHubIcon from '../icons/GitHubIcon';

class About extends Component {
    state = {}
    render() {
        return (
            <div className='about'>
                <div className='title'>Little Mathematician</div>
                <div className='content'>
                    Simple Maths game for children made in ReactJS with aim to explore this framework.
                </div>
                <div className='git-source'>
                    <GitHubIcon/><a href="https://github.com/LukaszGrela/little-mathematician-reacjs" 
                    target='_blank' 
                    rel="noopener noreferrer">LukaszGrela/little-mathematician-reacjs</a>
                </div>
                <div className='acknowledgement'>
                    <ul>
                        <li>Icons in menu (and logo) by Leona Grande from the Noun Project, thenounproject.com/</li>
                        <li>Other icons came from material ui icons, www.materialui.co/</li>
                    </ul>
                </div>
            </div>
        );
    }
}

export default About;
