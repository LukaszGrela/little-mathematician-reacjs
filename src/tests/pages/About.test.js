import React from 'react';
import { shallow } from 'enzyme';
import About from '../../pages/About';

test('Should render About page correctly', () => {
    const wrapper = shallow(<About />);

    expect(wrapper).toMatchSnapshot();
});