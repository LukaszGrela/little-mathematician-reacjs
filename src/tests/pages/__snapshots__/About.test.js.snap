// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render About page correctly 1`] = `
<div
  className="about"
>
  <div
    className="title"
  >
    Little Mathematician
  </div>
  <div
    className="content"
  >
    Simple Math game for children made in ReactJS to learn this framework.
  </div>
  <div
    className="git-source"
  >
    <GitHubIcon />
    <a
      href="https://github.com/LukaszGrela/little-mathematician-reacjs"
      rel="noopener noreferrer"
      target="_blank"
    >
      LukaszGrela/little-mathematician-reacjs
    </a>
  </div>
  <div
    className="acknowledgement"
  >
    <ul>
      <li>
        Icons in menu (and logo) by Leona Grande from the Noun Project, thenounproject.com/
      </li>
      <li>
        Other icons came from material ui icons, www.materialui.co/
      </li>
    </ul>
  </div>
</div>
`;
